lib/please_cli/please_cli/deploy.py

```diff
@@ -472,20 +478,25 @@ def cmd_TASKCLUSTER_HOOK(ctx,
         image = hook.get('task', {}).get('payload', {}).get('image', '')
         if image.startswith('/nix/store'):
 
-            image_tag = '-'.join(reversed(image[11:-7].split('-', 1)))
-            click.echo(' => Uploading docker image `{}:{}` ... '.format(docker_repo, image_tag), nl=False)
-            with click_spinner.spinner():
-                please_cli.utils.push_docker_image(
-                    registry=docker_registry,
-                    username=docker_username,
-                    password=docker_password,
-                    image=f'docker-archive://{image}',
-                    repo=docker_repo,
-                    tag=image_tag,
-                    interactive=interactive,
-                )
+            versioned_image_tag = '-'.join(reversed(image[11:-7].split('-', 1)))
+            image_tags = [versioned_image_tag]
+            if docker_stable_tag:
+                image_tags.append(docker_stable_tag)
+
+            for image_tag in image_tags:
+                click.echo(' => Uploading docker image `{}:{}` ... '.format(docker_repo, image_tag), nl=False)
+                with click_spinner.spinner():
+                    please_cli.utils.push_docker_image(
```

Comment:
Does this mean having to do two uploads?

Reply:
No docker is smart, and reuse the same chunks already uploaded